recipes/kwidgetsaddons/all/conanfile.py

```diff
@@ -0,0 +1,76 @@
+import os
+from conans import ConanFile, CMake, tools
+
+
+class KwidgetsaddonsConan(ConanFile):
+    name = "kwidgetsaddons"
+    license = ("BSD-3-Clause", "LGPL-2.0-only", "LGPL-2.1-only", "LGPL-3.0-only", "LicenseRef-KDE-Accepted-LGPL", 
+        "GPL-2.0-or-later", "LGPL-2.0-or-later", "LGPL-2.1-or-later", "LGPL-3.0-or-later")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://api.kde.org/frameworks/kwidgetsaddons/html/index.html"
+    topics = ("conan", "desktop", "widgets", "qt")
+    description = "kwidgetsaddons contains add-on widgets and classes for applications that use the Qt Widgets module."
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False]
+    }
+    default_options = {"shared": False, "fPIC": True}
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake"
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def build_requirements(self):
+        self.build_requires("extra-cmake-modules/5.75.0")
+
+    def requirements(self):
+        self.requires("qt/5.15.1")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("kwidgetsaddons-{}".format(self.version), self._source_subfolder)
+
+        tools.replace_in_file(
+            os.path.join(self._source_subfolder, "src", "CMakeLists.txt"),
+            "KDE_INSTALL_DATADIR_KF5",
+            "CONAN_KDE_INSTALL_DATADIR_KF5"
+        )
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+
+        self._cmake = CMake(self)
+        self._cmake.definitions['ECM_MKSPECS_INSTALL_DIR'] = os.path.join(self.package_folder, "res")
+        self._cmake.definitions['KDE_INSTALL_LOGGINGCATEGORIESDIR'] = os.path.join(self.package_folder, "res")
+        self._cmake.definitions['CONAN_KDE_INSTALL_DATADIR_KF5'] = os.path.join(self.package_folder, "res")
+        self._cmake.definitions['CMAKE_INSTALL_LOCALEDIR'] = os.path.join(self.package_folder, "res", "locale")
+        # a hack to avoid installing CMake find modules andconfig files (KB-H016)
+        self._cmake.definitions['KDE_INSTALL_CMAKEPACKAGEDIR'] = os.path.join(self.build_folder, "dummy")
```

Comment:
Can you check the name of the generated cmake files + targets and recreate them in `package_info`?

Reply:
I tried this:
```
        self.cpp_info.name = "KF5"
        self.cpp_info.components["WidgetsAddons"].libs = ["KF5WidgetsAddons"]
        self.cpp_info.components["WidgetsAddons"].includedirs = ["include/KF5", "include/KF5/KWidgetsAddons"]
        self.cpp_info.components["WidgetsAddons"].names["cmake_find_package"] = "WidgetsAddons"
        self.cpp_info.components["WidgetsAddons"].names["cmake_find_package_multi"] = "WidgetsAddons"
```

Which can be linked in cmake using
```
find_package(KF5 5.75.0 REQUIRED COMPONENTS WidgetsAddons)
target_link_libraries(example KF5::kwidgetsaddons)
```

However this break conan TARGETS mode as I cannot do this anymore
```
target_link_libraries(example CONAN_PKG::kwidgetsaddons)
```
because the name is now KF5  instead of kwidgetsaddons.

Not sure how to fix this. Ultimately we should have 
* **CONAN_PKG::kwidgetsaddons** in conan target mode
*  **KF5::kwidgetsaddons** in cmake find_package mode

Reply:
Is it this what you want?
```python
self.cpp_info.names["cmake_find_package"] = "KF5"
self.cpp_info.names["cmake_find_package_multi"] = "KF5"
self.cpp_info.components["libkwidgetsaddons"].libs = ["KF5WidgetsAddons"]
self.cpp_info.components["libkwidgetsaddons"].includedirs = ["include/KF5", "include/KF5/KWidgetsAddons"]
self.cpp_info.components["libkwidgetsaddons"].names["cmake_find_package"] = "WidgetsAddons"
self.cpp_info.components["libkwidgetsaddons"].names["cmake_find_package_multi"] = "WidgetsAddons"
self.cpp_info.components["libkwidgetsaddons"].requirse = ["qt5::qt5"]  # FIXME: should probably only require "qt5::widgets"
self.cpp_info.components["_lib2"].names["cmake_find_package"] = "kwidgetsaddons"
self.cpp_info.components["_lib2"].names["cmake_find_package_multi"] = "kwidgetsaddons"
```

This enables:
```cmake
find_package(KF5 REQUIRED COMPONENTS WidgetsAddons)
target_link_libraries(example KF5::kwidgetsaddons) 
# target_link_libraries(example KF5::WidgetsAddons) is also possible, but that is a limitation of the imperfect conan modelling of the components.

# This still works
target_link_libraries(example CONAN_PKG::kwidgetsaddons)
```